### PR TITLE
[ticket/13147] Change get_global_visibility_sql()'s results

### PR DESCRIPTION
**Pull request by brunoais**
https://tracker.phpbb.com/browse/PHPBB3-13147

fixing pushing branch on #3200

### CODE REVIEW

File: tests/content_visibility/get_visibility_sql_test.php [hunk 1]
```diff
@@ -83,7 +83,8 @@ public function test_get_visibility_sql($table, $mode, $forum_id, $table_alias,
 			->will($this->returnValueMap($permissions));
 		$user = new \phpbb\user('\phpbb\datetime');
 		$config = new phpbb\config\config(array());
-		$content_visibility = new \phpbb\content_visibility($auth, $config, $db, $user, $phpbb_root_path, $phpEx, FORUMS_TABLE, POSTS_TABLE, TOPICS_TABLE, USERS_TABLE);
+		$phpbb_dispatcher = $this->getMock('\phpbb\event\dispatcher');
```
Comment (nickvergessen) [inline tests/content_visibility/get_visibility_sql_test.php]:
```php
$this->getMockBuilder('\phpbb\event\dispatcher')
	->disableOriginalConstructor()
	->getMock();
```

File: tests/content_visibility/set_topic_visibility_test.php [hunk 2]
```diff
@@ -90,7 +90,8 @@ public function test_set_topic_visibility($visibility, $topic_id, $forum_id, $us
 		$auth = $this->getMock('\phpbb\auth\auth');
 		$user = new \phpbb\user('\phpbb\datetime');
 		$config = new phpbb\config\config(array());
-		$content_visibility = new \phpbb\content_visibility($auth, $config, $db, $user, $phpbb_root_path, $phpEx, FORUMS_TABLE, POSTS_TABLE, TOPICS_TABLE, USERS_TABLE);
+		$phpbb_dispatcher = $this->getMock('\phpbb\event\dispatcher');
```
Comment (nickvergessen) [inline tests/content_visibility/set_topic_visibility_test.php]:
```php
$this->getMockBuilder('\phpbb\event\dispatcher')
	->disableOriginalConstructor()
	->getMock();
```

**Comment by brunoais** [comment]
!unset WIP

**Comment by brunoais** [comment]
!unset WIP

File: phpBB/phpbb/content_visibility.php [hunk 3]
```diff
@@ -244,6 +244,34 @@ public function get_global_visibility_sql($mode, $exclude_forum_ids = array(), $
 
 		$approve_forums = array_diff(array_keys($this->auth->acl_getf('m_approve', true)), $exclude_forum_ids);
 
+		$get_global_visibility_sql_overwrite = null;
+		/**
```
Comment (nickvergessen) [inline phpBB/phpbb/content_visibility.php]: new line please before the doc block

File: phpBB/phpbb/content_visibility.php [hunk 4]
```diff
@@ -244,6 +244,34 @@ public function get_global_visibility_sql($mode, $exclude_forum_ids = array(), $
 
 		$approve_forums = array_diff(array_keys($this->auth->acl_getf('m_approve', true)), $exclude_forum_ids);
 
+		$get_global_visibility_sql_overwrite = null;
```
Comment (nickvergessen) [inline phpBB/phpbb/content_visibility.php]: I'd remove `get_global_` from the variable name, so it's a bit shorter

**Pull request merged**